### PR TITLE
cleanup(mixin): add integration tests

### PR DESCRIPTION
**Pull request by cuiy0006**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14829)

### CODE REVIEW

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:134]: can we move this to its own `TEST_F(...)`?

similar to https://github.com/googleapis/google-cloud-cpp/blob/291074bca1fb36966696db56f1d20f16911fe327/google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc#L657

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:136]: can we use `GetOperation` instead of `ListOperations` ? something like:

```cc
TEST_F(DatabaseAdminClientTest, MixinAPI) {
  // Start an LRO.
  auto operation = client_.CreateDatabase(
      NoAwaitTag{}, database_.instance().FullName(),
      absl::StrCat("CREATE DATABASE `", database_.database_id(), "`"));
  ASSERT_STATUS_OK(operation);

  // Call a mixin API
  auto get_operation = client_.GetOperation(operation->name());
  ASSERT_STATUS_OK(get_operation);

  // Verify mixin results
  EXPECT_EQ(get_operation->name(), operation->name()); // or something like this, idk.

  // Wait for the LRO to complete, then clean up the test artifacts
  (void)client_.CreateDatabase(*operation).get();
  (void)client_.DropDatabase(database_.FullName());
}
```

Comment (cuiy0006) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:136]: Oh, I see, there is `NoAwaitTag`.

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc
Comment (cuiy0006) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc]: Done.

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc
Comment (cuiy0006) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc]: Done.

**Review by dbolduc — APPROVED** [review]
Please wait until after the release to merge

**Pull request merged**
